conduit/data/datasets/vision/waterbirds.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+from pathlib import Path
+from typing import ClassVar, Optional, Union, cast
+
+from kit.misc import str_to_enum
+import pandas as pd
+import torch
+
+from conduit.data.datasets.utils import ImageTform, UrlFileInfo, download_from_url
+from conduit.data.datasets.vision.base import CdtVisionDataset
+
+__all__ = ["WaterbirdsDataset", "WaterbirdsSplit"]
+
+
+class WaterbirdsSplit(Enum):
+    train = 0
+    val = 1
+    test = 2
+
+
+class WaterbirdsDataset(CdtVisionDataset):
+    """The Waterbirds dataset.
+    The dataset was constructed from the CUB-200-2011 dataset and the Places dataset:
+    """
+
+    _BASE_FOLDER: ClassVar[str] = "Waterbirds"
+    _FILE_INFO: ClassVar[UrlFileInfo] = UrlFileInfo(
+        name="Waterbirds.tar.gz",
+        url="https://worksheets.codalab.org/rest/bundles/0x505056d5cdea4e4eaa0e242cbfe2daa4/contents/blob/",
+        md5=None,
+    )
+
+    def __init__(
+        self,
+        root: Union[str, Path],
+        *,
+        download: bool = True,
+        transform: Optional[ImageTform] = None,
+        split: Optional[Union[WaterbirdsSplit, str]] = None,
+    ) -> None:
+
+        if isinstance(split, str):
+            split = str_to_enum(str_=split, enum=WaterbirdsSplit)
+
+        self.root = Path(root)
+        self._base_dir = self.root / self._BASE_FOLDER
+        self.download = download
+        if self.download:
+            download_from_url(
+                file_info=self._FILE_INFO,
+                root=self.root,
+                logger=self.logger,
+                remove_finished=True,
+            )
+        else:
+            raise FileNotFoundError(
```

Comment:
Can this check be somehow be put into shared code?

Reply:
yes, I was thinking the same thing; this else block is repeated an awful lot. Nonetheless, I'm not going to do it in this PR as it's a fairly time-consuming change to make.